### ISSUE TITLE
NullReferenceException when charging to full

### ISSUE THREAD

**Issue by mgrddsj** [myfix16/BatteryTracker#1]
I have been using this application for quite some time, and have encountered this multiple times. I observed that whenever my laptop battery charges from 99% to 100%, a .NET Framework error window will show up, stating that there's a NullReferenceException. I'm using the prebuilt binary in v1.0 release. 

![image](https://user-images.githubusercontent.com/36990144/114641334-8911b080-9d04-11eb-8913-18fd194dedbb.png)

The log is here: 
https://gist.github.com/mgrddsj/acf4a79202fe5efbc171844f0670af63

**Comment by myfix16** [comment]
I used the same release for quite a long time and it works well for me. Could you describe more details of this bug? Does it happen every time your battery life goes from 99% to 100% or just sometimes? If you click continue, will it work normally?

**Comment by mgrddsj** [comment]
It happens sometimes, I tried unplugging my power cord and plugging it in again just now, and it doesn't pop up. But it happens quite often in my daily use. It seems to happen by chance. 

When I click continue, the app will work normally as if nothing happened. Maybe a simple try...catch statement will mitigate the issue? (though it's not a good practice to do so.)

**Comment by mgrddsj** [comment]
I found that if I unplug and plug in immediately, it won't cause the exception, as it becomes full instantly if I plug in at battery >95%. However, when I tried draining my battery below 95% and slowly charge it up to 100%, the exception appears. 

I noticed that my battery seems to "make a feint shot" and reports that it's fully charged, and sometimes it may drop back to 99% again (instantly). This phenomenon happens more often when I use a 100W PD charger instead of the 230W power brick that comes with the laptop (but sometimes the application throws the exception too when I use the 230W power brick). 

Charging also becomes noticeably slower as it gets closer to 100%, it took about 5 minutes to charge from 99% to 100% in my brief experiment. Charging rate also became lower.

https://user-images.githubusercontent.com/36990144/114651313-4eb10f00-9d16-11eb-87e6-bb9737edf663.mp4

Also, when it jumps to 100% for a second, the reported battery percentage on BatteryInfoView did not change. It becomes really full after a few more minutes, and this it what it looks like when it's really fully charged: 

![image](https://user-images.githubusercontent.com/36990144/114652617-d39d2800-9d18-11eb-8b4e-fcb31cb3320c.png)
